x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -71,19 +78,53 @@ public static List<ValueSourceInfo> sources(
             try {
                 fieldData = ctx.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
             } catch (IllegalArgumentException e) {
-                if (asUnsupportedSource) {
-                    sources.add(
+                switch (elementType) {
+                    case BYTES_REF -> sources.add(
                         new ValueSourceInfo(
                             new UnsupportedValueSourceType(fieldType.typeName()),
                             new UnsupportedValueSource(null),
                             elementType,
                             ctx.getIndexReader()
                         )
                     );
-                    continue;
-                } else {
-                    throw e;
+
+                    case LONG, INT -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.NUMERIC, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
+                    );
+
+                    case BOOLEAN -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.BOOLEAN, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
```

Comment:
Question: why is the numeric values source correct for an unsupported Boolean source?

Reply:
I think `booleans` representation is the same as `byte`, that is a numeric